src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/SolutionExplorer/PackageManagerUICommandHandler.cs

```diff
@@ -308,33 +306,24 @@ private async Task<vsShellInterop.IVsWindowFrame> CreateNewWindowFrameAsync(stri
             var hr = 0;
             int[] pfDefaultPosition = null;
 
-            try
-            {
-                hr = uiShell.CreateToolWindow(
-                    (uint)__VSCREATETOOLWIN.CTW_fInitNew,
-                    0,              // dwToolWindowId - singleInstance = 0
-                    windowPane,     // ToolWindowPane
-                    Guid.Empty,     // rclsidTool = GUID_NULL
-                    Guid.NewGuid(), // TODO: should be projectGuid...so persistance info works.
-                    Guid.Empty,     // reserved - do not use - GUID_NULL
-                    null,           // IServiceProvider
-                    caption,
-                    pfDefaultPosition,
-                    out windowFrame);
-
-                if (windowFrame != null)
-                {
-                    WindowFrameHelper.AddF1HelpKeyword(windowFrame, keywordValue: F1KeywordValuePmUI);
-                    WindowFrameHelper.DisableWindowAutoReopen(windowFrame);
-                    WindowFrameHelper.DockToolWindow(windowFrame);
-                }
-            }
-            finally
+            var windowPane = new PackageManagerToolWindowPane(control);
+            hr = uiShell.CreateToolWindow(
+                (uint)__VSCREATETOOLWIN.CTW_fInitNew,
+                0,              // dwToolWindowId - singleInstance = 0
+                windowPane,     // ToolWindowPane
+                Guid.Empty,     // rclsidTool = GUID_NULL
+                Guid.NewGuid(), // TODO: should be projectGuid...so persistance info works.
+                Guid.Empty,     // reserved - do not use - GUID_NULL
+                null,           // IServiceProvider
+                caption,
+                pfDefaultPosition,
+                out windowFrame);
+
+            if (windowFrame != null)
             {
-                if (windowPane != null)
-                {
-                    windowPane.Dispose();
```

Comment:
Is this the cause of the Suppression for not disposing? If so, should that be global or should it be just on this method?

Reply:
I think it's OK to have it just here for now, but definitely something we should move to the global list if we create more windows.